session.go

```diff
@@ -407,7 +413,22 @@ func (s *Session) handleAckFrame(frame *frames.AckFrame) error {
 }
 
 // Close the connection. If err is nil it will be set to qerr.PeerGoingAway.
+// It waits until the run loop has stopped before returning
 func (s *Session) Close(e error) error {
+	err := s.closeImpl(e, false)
+
+	if atomic.LoadUint32(&s.closed) == 1 {
```

Comment:
Right. I was trying to catch multiple calls to `Close` there, but it doesn't work that way.
I added a fix, that now works with an explicit error returned by the `closeImpl` in case the session is already closed.